lib/core.js

```diff
@@ -307,6 +307,15 @@ interface Iterable<T> {
     @@iterator(): Iterator<T>;
 }
 
+/* Generators */
+interface Generator<Y,R,N> {
+    @@iterator(): Iterator<Y>;
+    next(value?: N): IteratorResult<Y|R>;
+}
+
```

Comment:
Sorry, I'm a bit unclear about what that buys us. Could you extend your example with a test case showing an expected type error?